File: 01-ds/linkedlist.go
package main

import "fmt"

type Node struct {
	data int
	nPtr *Node
}

type LinkedList struct {
	head *Node
	next *Node
}

func (l *LinkedList) addNode(data int) {
	if l.head == nil {
		l.head = &Node{data: data}
		l.next = l.head
	} else {
		l.next.nPtr = &Node{data: data}
		l.next = l.next.nPtr
	}
}

func main() {
	fmt.Println("Hello World")
}

File: javascript/LinkedList.js
class LinkedList {
    constructor(node) {
        this.head = node;
        this.tail = null;
    }

    append(node) {
        if (this.head === null) {
            this.head = node;
            return;
        }
    }
    
    remove(value) {
        if (this.head === null) {
            return;
        }
        if (this.head.value === value) {
            this.head = this.head.next;
            return;
        }
    }
    search(value) {
        if (this.head === null) {
            return "Not found";
        }

        for (let current = this.head; current.value !== null; current = current.next) {
            if (current.value === value) {
                return "Found";
            }
        }

    }

}

class Node {
    constructor(value) {
        this.value = value;
        this.next = null;
    }
}
